lib/spack/spack/build_environment.py

```diff
@@ -525,22 +525,29 @@ def child_fun():
     carries on.
     """
 
-    def child_execution(child_connection):
+    def child_execution(child_connection, input_stream):
         try:
             setup_package(pkg, dirty=dirty)
-            function()
+            function(input_stream)
             child_connection.send([None, None, None])
         except Exception as e:
             child_connection.send([type(e), e, None])
         finally:
             child_connection.close()
 
     parent_connection, child_connection = multiprocessing.Pipe()
-    p = multiprocessing.Process(
-        target=child_execution,
-        args=(child_connection,)
-    )
-    p.start()
+    try:
+        # Forward sys.stdin to be able to activate / deactivate
+        # verbosity pressing a key at run-time
+        input_stream = os.fdopen(os.dup(sys.stdin.fileno()))
```

Comment:
Ok I'm a little confused.  Why do we need to forward stdin to the child to get keypresses working?  The parent should be receiving output from the child and input from the user, but the user's terminal is already hooked up to the parent.  So why does it need to be forwarded to the child and closed on the parent?